api/errors.go

```diff
@@ -10,39 +11,96 @@ import (
 // The canonical errors from the EigenDA gRPC API endpoints.
 //
 // Notes:
-// - We start with a small (but sufficient) subset of grpc's error codes,
-//   and expand when there is an important failure case to separate out. See:
-//   https://grpc.io/docs/guides/status-codes/
-// - Make sure that internally propagated errors are eventually wrapped in one of the
-//   user-facing errors defined here, since grpc otherwise returns an UNKNOWN error code,
-//   which is harder to debug and understand for users.
-// - See https://github.com/googleapis/googleapis/blob/ba8ea80f25d19bde8501cd51f314391f8d39bde8/google/rpc/code.proto
-//   for the mapping of grpc error codes to HTTP status codes.
-
+//   - We start with a small (but sufficient) subset of grpc's error codes,
+//     and expand when there is an important failure case to separate out. See:
+//     https://grpc.io/docs/guides/status-codes/
+//   - Make sure that internally propagated errors are eventually wrapped in one of the
+//     user-facing errors defined here, since grpc otherwise returns an UNKNOWN error code,
+//     which is harder to debug and understand for users.
+//   - See https://github.com/googleapis/googleapis/blob/ba8ea80f25d19bde8501cd51f314391f8d39bde8/google/rpc/code.proto
+//     for the mapping of grpc error codes to HTTP status codes.
 func newErrorGRPC(code codes.Code, msg string) error {
 	return status.Error(code, msg)
 }
 
+// WrapAsGRPCError wraps an error with a gRPC code and message. If the error already has a gRPC code,
+// it retains the code but combines the messages. Otherwise, it uses the provided code and combined message.
+func WrapAsGRPCError(err error, code codes.Code, msg string) error {
+	if err == nil {
+		return nil
```

Comment:
shouldnt this just use newErrorGRPC in this case?

Reply:
explained above

Reply:
Don't really like this semantic though, would prefer at each call site the more explicit
```
if err != nil {
   WrapAsGRPCError(err, code, msg)
}
```
instead of wrapping by default and forcing reader to go make sure that the function returns nil if no error was present

Reply:
its not defensive coding... its the wrong semantics.